xapian-letor/api/letor_internal.cc

```diff
@@ -123,16 +113,18 @@ Letor::Internal::load_list_fvecs(const char *filename) { //directly use train.tx
         train_file >> qid;
 
         qid = qid.substr(qid.find(":")+1,qid.length());
+        train_file.ignore();
 
-        std::map<int,double> fvals;//read features
-        for(int i = 1; i < 20; ++i){
-            train_file.ignore();
+        std::vector<double> fvals;
+
+        while (train_file.peek() != 35) { // read till '#docid' is found (35 is ascii for '#')
```

Comment:
Pretty sure `'#'` (`char`) will upcast to `int` happily for this comparson. The comment is still needed to point out we're looking for `#docid`, though.